### PR TITLE
feat: implement max marginal relevance for momento vector index

### PR DESCRIPTION
**Pull request by malandis**
**Description**

Implements max marginal relevance search for the Momento Vector Index vector store integration.

**Dependencies**

Bumps the `@gomomento` dependencies to get support for `searchAndFetchVectors`, which is needed for the MMR implementation.

**Tests**

✅ adds integration test for MMR to verify diversity-based ranking

**Twitter handle**

Please shoutout @momentohq for Momento Vector Index and @mlonml for the contribution. With this contribution, MVI now supports all core vector store methods!

### CODE REVIEW

File: examples/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "@elastic/elasticsearch": "^8.4.0",
     "@getmetal/metal-sdk": "^4.0.0",
     "@getzep/zep-js": "^0.9.0",
-    "@gomomento/sdk": "^1.47.1",
+    "@gomomento/sdk": "^1.51.1",
```
Comment (malandis) [inline examples/package.json:29]: Bumped in the example to get `searchAndFetchVectors` support.

Comment (malandis) [inline examples/package.json:29]: Bumped to get `searchAndFetchVectors` implementation in momento client.

File: langchain/package.json [hunk 2]
```diff
@@ -856,8 +856,8 @@
     "@faker-js/faker": "^7.6.0",
     "@getmetal/metal-sdk": "^4.0.0",
     "@getzep/zep-js": "^0.9.0",
-    "@gomomento/sdk": "^1.47.1",
-    "@gomomento/sdk-core": "^1.47.1",
+    "@gomomento/sdk": "^1.51.1",
+    "@gomomento/sdk-core": "^1.51.1",
```
Comment (malandis) [inline langchain/package.json:860]: Bumped to get `searchAndFetchVectors` support in Momento client. `@gomomento/sdk` has the actual client implementation (needed for integration tests) and `sdk-core` for the interface and message types.

File: langchain/package.json [hunk 3]
```diff
@@ -993,9 +993,9 @@
     "@elastic/elasticsearch": "^8.4.0",
     "@getmetal/metal-sdk": "*",
     "@getzep/zep-js": "^0.9.0",
-    "@gomomento/sdk": "^1.47.1",
-    "@gomomento/sdk-core": "^1.47.1",
-    "@gomomento/sdk-web": "^1.47.1",
+    "@gomomento/sdk": "^1.51.1",
+    "@gomomento/sdk-core": "^1.51.1",
+    "@gomomento/sdk-web": "^1.51.1",
```
Comment (malandis) [inline langchain/package.json:998]: Bumped to get `searchAndFetchVectors` support in Momento client. `@gomomento/sdk` has the actual client implementation for Node.js; `@gomomento/sdk-web` for the browser client; and `sdk-core` for the interface and message types (needed to get the project to build).

**Review by malandis — COMMENTED** [review]
Left some clarifying comments about momento version numbers.

**Comment by jacoblee93** [comment]
LGTM, thank you!

**Pull request merged**